data/publisher/src/main/java/tech/pegasys/teku/data/publisher/SystemMetricData.java

```diff
@@ -13,57 +13,220 @@
+
+  @JsonProperty("memory_node_bytes_cached")
+  private final long memoryNodeBytesCached = 0L;
+
+  @JsonProperty("memory_node_bytes_buffers")
+  private final long memoryNodeBytesBuffers = 0L;
+
+  @JsonProperty("disk_node_bytes_total")
+  private long diskNodeBytesTotal = 0L;
+
+  @JsonProperty("disk_node_bytes_free")
+  private long diskNodeBytesFree = 0L;
+
+  @JsonProperty("disk_node_io_seconds")
+  private final long diskNodeIoSeconds = 0L;
+
+  @JsonProperty("disk_node_reads_total")
+  private final long diskNodeReadsTotal = 0L;
+
+  @JsonProperty("disk_node_writes_total")
+  private final long diskNodeWritesTotal = 0L;
+
+  @JsonProperty("network_node_bytes_total_receive")
+  private final long networkNodeBytesTotalReceive;
+
+  @JsonProperty("network_node_bytes_total_transmit")
+  private final long networkNodeBytesTotalTransmit;
+
+  @JsonProperty("misc_node_boot_ts_seconds")
+  private final long miscNodeBootTsSeconds;
+
+  @JsonProperty("misc_os")
+  private final String miscOs = getNormalizedOSVersion();
+
+  public SystemMetricData(long timestamp, final File beaconNodeDirectory) {
+    super(timestamp, SYSTEM.getDisplayName());
+
+    SystemInfo systemInfo = new SystemInfo();
+    final HardwareAbstractionLayer hardware = systemInfo.getHardware();
+    hardware.getProcessor().getPhysicalProcessorCount();
+    this.cpuCores = hardware.getProcessor().getPhysicalProcessorCount();
+    this.cpuThreads = hardware.getProcessor().getLogicalProcessorCount();
+    this.memoryNodeBytesFree = hardware.getMemory().getAvailable();
+    this.memoryNodeBytesTotal = hardware.getMemory().getTotal();
+    this.miscNodeBootTsSeconds = getNodeBootSeconds();
+
+    final long[] processStats =
+        setProcessLoadTicks(hardware.getProcessor().getProcessorCpuLoadTicks());
+    this.cpuNodeUserSecondsTotal = processStats[0];
+    this.cpuNodeSystemSecondsTotal = processStats[1];
+    this.cpuNodeIdleSecondsTotal = processStats[2];
+    this.cpuNodeIowaitSecondsTotal = processStats[3];
+
+    this.networkNodeBytesTotalReceive =
+        hardware.getNetworkIFs(false).stream().map(NetworkIF::getBytesRecv).reduce(0L, Long::sum);
+    this.networkNodeBytesTotalTransmit =
+        hardware.getNetworkIFs(false).stream().map(NetworkIF::getBytesSent).reduce(0L, Long::sum);
+
+    this.diskNodeBytesTotal =
+        beaconNodeDirectory.exists() ? beaconNodeDirectory.getTotalSpace() : 0L;
+    this.diskNodeBytesFree = beaconNodeDirectory.exists() ? beaconNodeDirectory.getFreeSpace() : 0L;
+  }
+
+  private long getNodeBootSeconds() {
+    switch (miscOs) {
+      case "lin":
+        return new LinuxOperatingSystem().getSystemBootTime();
+      case "win":
+        return new WindowsOperatingSystem().getSystemBootTime();
+      case "mac":
+        return new MacOperatingSystem().getSystemBootTime();
+      default:
+        return 0L;
+    }
+  }
+
+  private long[] setProcessLoadTicks(final long[][] processorCpuLoadTicks) {
```

Comment:
This doesn't appear to set anything.  Maybe should be `calculateProcessLoadTicks`?

Reply:
fair point, name is a victim of refactoring.. will change.